venvs/tests/test_make.py

```diff
@@ -140,6 +140,44 @@ def test_multiple_installs_one_link(self):
     def test_multiple_installs_one_link_no_name(self):
         self.run_cli(["-i", "foo", "-i", "bar", "-l", "foo"], exit_status=2)
 
+    def test_install_edit_config(self):
+        """Install will automatically edit the config file."""
+        self.run_cli(["-l", "foo", "-i", "bar", "--persist"])
+        contents = _load_config(
+            filesystem=self.filesystem,
+            locator=self.locator,
+        )
+        self.assertEqual(
+            contents,
+            {'virtualenv': {"bar": {"install": ["bar"], "link": ["foo"]}}}
+        )
+
+    def test_handle_empty_config_file(self):
+        """Don't'break with an empty config file."""
+
+        # Empty the config file.
+        with self.filesystem.open(
+                self.locator.root.descendant("virtualenvs.toml"),
+                "w",
+        ):
+            pass
+
+        self.run_cli(["-l", "foo", "-i", "bar", "--persist"])
+        contents = _load_config(
+            filesystem=self.filesystem,
+            locator=self.locator,
+        )
+        self.assertEqual(
+            contents,
+            {'virtualenv': {"bar": {"install": ["bar"], "link": ["foo"]}}}
+        )
+
+    def test_install_no_persist(self):
+        """Install --no-persist will not edit the config file."""
+        self.run_cli(["-l", "foo", "-i", "bar", "--no-persist"])
+        with self.assertRaises(filesystems.exceptions.FileNotFound):
+            _load_config(filesystem=self.filesystem, locator=self.locator)
```

Comment:
Hm, this is odd though -- this test passes? Why do you get file-related errors when you're not asking venvs to edit a file?

Reply:
The idea of this test is that no file has been created, so it should fail with `FileNotFound`. cfca4de7d882efc83759025a4f0fd17fc8732b52 adds a comment to clarify. The pathlib way would be `assert not pathlib.Path(pth).exists()`. I didn't see an `.exists()` method in `filesystems`.